pkg/lib/analytic/posthog.go

```diff
@@ -149,6 +154,21 @@ func (p *PosthogIntegration) preparePosthogGroup(ctx context.Context, appID stri
 		return nil, err
 	}
 
+	var m2mTokenCreatedCount int
+	err = p.AuditDBReadHandle.WithTx(ctx, func(ctx context.Context) error {
+		rangeTo := now
+		rangeFrom := now.AddDate(0, 0, -30)
```

Comment:
For consistency, we should add a new function in timeutil `PreviousMonthRange`

```
// PreviousMonthRange is like PreviousMonth, except that it returns a end-exclusive range representing the result.
func PreviousMonthRange(t time.Time) (rangeFrom time.Time, rangeTo time.Time)
```

Reply:
Added a commit for this, I named it Last30Days because it seems to be more accurate.

Reply:
No, I really meant PreviousMonthRange. So the collected metrics share the same time span.